observableeditablebuffer.go

```diff
@@ -0,0 +1,209 @@
+package main
+
+import "fmt"
+
+type ObservableEditableBuffer interface {
+	AddText(observer BufferObserver)
+	DelText(observer BufferObserver) error
+	SetCurText(observer BufferObserver)
+	GetCurText() interface{}
+	GetTextSize() int
+	AllText(tf func(i interface{}))
+	HasMultipleTexts() bool
+	InsertAt(p0 int, s []rune)
+	InsertAtWithoutCommit(p0 int, s []rune)
+	DeleteAt(p0, p1 int)
+	Undo(isundo bool) (q0, q1 int, ok bool)
+	New() *Editbuf
+}
+
+type Editbuf struct {
+	curtext BufferObserver
+	text    map[BufferObserver]struct{} // [private I think]
+}
+
+func (f *File) AddText(observer BufferObserver) {
```

Comment:
You've sort of started following #334 but not quite? Before continuing, please common on #334 to explain the type relationship. 

Remember that the goal is to section the original `File` into:

* "the part that is like `undo.Buffer` (call this `File` for now)
* "the rest" (`ObservableEditableBuffer`)

The `AddText` (now should presumably be renamed to `AddObserver`?) is a method on `ObservableEditableBuffer` while (for example) `InsertAt` stays on `File`. (Which you have here but it should then be in the `file.go` file.)

Reply:
Added comments to #334